Penumbra/UI/ModsTab/ModPanelTagsFeatureTab.cs

```diff
@@ -0,0 +1,211 @@
+    private readonly ModFileSystemSelector _selector;
+    private readonly TutorialService _tutorial;
+    private readonly ModManager _modManager;
+    private readonly TagButtons _localTags = new();
+
+    //Contains the tags
+    private List<List<string>> _tagsList = new();
+    //private Dictionary<string, bool> _tagDict = new();
+    //private List<string> _selectedTags = new();
+
+    public ModPanelTagsFeatureTab(ModFileSystemSelector selector, TutorialService tutorial, ModManager modManager)
+    {
+        _selector = selector;
+        _tutorial = tutorial;
+        _modManager = modManager;
+
+        AddTagsToList();
+    }
+
+    public ReadOnlySpan<byte> Label
+        => "Tag Test Feature"u8;
+
+    public void DrawContent()
+    {
+        //Maybe remove this from the description tab?
+        var tagIdx = _localTags.Draw("Local Tags: ",
+            "Custom tags you can set personally that will not be exported to the mod data but only set for you.\n"
+          + "If the mod already contains a local tag in its own tags, the local tag will be ignored.", _selector.Selected!.LocalTags,
+            out var editedTag, false);
+        _tutorial.OpenTutorial(BasicTutorialSteps.Tags);
+        if (tagIdx >= 0)
+            _modManager.DataEditor.ChangeLocalTag(_selector.Selected!, tagIdx, editedTag);
+
+
+        //Adds the buttons, right now the tags are just in a List that contains List<string>, I'd properly move the tags to a json file so its easier to add and remove tags, can also add a function so the users can add tags themselves
+        DrawButtons(_tagsList[0], "Type");
+        DrawButtons(_tagsList[1], "Genre");
+        DrawButtons(_tagsList[2], "Gear Types");
+        DrawButtons(_tagsList[3], "Gear Category");
+        DrawButtons(_tagsList[4], "Gear Styles");
+        DrawButtons(_tagsList[5], "Gear Material");
+
+    }
+
+
+    private void DrawButtons(List<string> tags, string title)
+    {
+        ImGui.Text(title);
+        //byte buttonsOnSameLine = 0;
+        foreach (var tag in tags)
+        {
+            //Checks if the mod does not have the tag
+            if (!_selector.Selected!.LocalTags.Contains(tag))
+            {
+                //Clicking the button adds the tag to the localTags list
+                if (ImGui.Button(tag))
+                {
+                    AddTag(tag, _selector.Selected!.LocalTags.Count);
+                }
+            }
+            //if it does we want to color the button so you can see its added, im pretty sure im using this wrong...?
+            else
+            {
+                var color = ImRaii.PushColor(ImGuiCol.Button, ColorId.SelectedCollection.Value());
+                //Clicking the button removes the tag from the localTags list
+                if (ImGui.Button(tag))
+                {
+                    RemoveTag(_selector.Selected!.LocalTags.IndexOf(tag));
+                }
+                color.Pop();
+            }
+            ImGui.SameLine();
```

Comment:
Currently there's horizontal scrolling involved for large categories. That should never be the case. Instead of fixing a number per line as implied by the commented out stuff, you would need to compute the size of the next tag in imgui, check the remaining horizontal space, and add a line break if it does not fit anymore. 
See my tag system for this, as it already does that somewhere.

Reply:
Rather use something else than buttons like I mentioned above but if you'd rather we use buttons for it then I'll fix that of course